[BUG]: `"composite" true` and Yarn workspace prevents `pgTable` from being compiled

### What version of `drizzle-orm` are you using?

0.26.1

### What version of `drizzle-kit` are you using?

0.18.1

### Describe the Bug

## TL;DR

TypeScript can't compile simple `schema.ts` file with `users` example table when using `"composite": true` in `tsconfig.json`

## Longer version

I have Yarn workspace with two packages, let's call them `backend` and `frontend`.
Naturally, `frontend` depends on `backend`. For that to work, `backend`'s `tsconfig.json` must include `"composite": true` option. 

`backend` uses `drizzle-orm` internally for migrations and I would like to use a standalone `schema.ts` file where all of the tables would be declared. With such setup, I would need to export said tables to use them elsewhere in the code. 

This is the moment where `export` before the `users` table declaration makes TypeScript compiler think that anyone can import `users` (type of which depends on inference of `drizzle-orm` package), even outside of the `backend` package. Due to lack of [`export(package)`](https://doc.rust-lang.org/reference/visibility-and-privacy.html) in TypeScript, this leads to compiler errors.

<details>
<summary>
schema.ts
</summary>

```ts
import { pgTable, serial, text, varchar } from "drizzle-orm/pg-core";

export const users = pgTable("users", {
  id: serial("id").primaryKey(),
  fullName: text("full_name"),
  phone: varchar("phone", { length: 256 }),
});
```

</details>

```logs
- error TS2742: The inferred type of 'users' cannot be named without a reference to '../../node_modules/drizzle-orm/db.d-2eb7c122'. This is likely not portable. A type annotation is necessary.

3 export const users = pgTable("users", {
               ~~~~~


Found 1 error.

error Command failed with exit code 1.
```

### Expected behavior

Everything compiles just fine.

I found possible workarounds, but none of them are optimal: https://github.com/microsoft/TypeScript/issues/47663#issuecomment-1519138189

### Environment & setup

_No response_

--- a/drizzle-orm/rollup.dts.config.ts
+++ b/drizzle-orm/rollup.dts.config.ts
@@ -13,6 +13,7 @@ export default defineConfig([
 		}, {}),
 		output: {
 			dir: 'dist.new',
+			minifyInternalExports: false,
 		},
 		external,
 		plugins: [

